packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1641,8 +1642,11 @@ export class AccountHandler {
           uid: account.uid,
         };
 
-        // a new key won't be autogenerated for users with 2FA enabled
-        if (hasTotpToken) {
+        // a new key won't be autogenerated for users with 2FA enabled or currently,
+        // for mobile users, due to the web view automatically closing after a
+        // successful login. The `isFirefoxMobileClient` option matches the
+        // client-side check against `integration.isFirefoxMobileClient()`.
+        if (hasTotpToken || isFirefoxMobileClient) {
```

Comment:
I was thinking it might be nicer to use something like `newAccountRecoveryKeyGenerated` since the method here doesn't need to know _why_ a new key isn't generated, but that doesn't play nicely with the need for `hasTotpToken` (at least, until we consider https://mozilla-hub.atlassian.net/browse/FXA-10894 ...)

Reply:
Yeah, this didn't feel super ideal. I've left a note in https://github.com/mozilla/fxa/compare/FXA-10815.